File: js/binarySearch.js

/*

Binary Search algorithm (JavaScript)
By CatProgrammerYT

*/

// Code below is not supossed to be editable, but if you need - you can change it as you want
let binarySearch = function (arr, number, position) { 

        // Start is the start position of the array (who could guess?)
	let start=0;
        // The end position (ye, cool)
        let end=arr.length-1; 

	while (start<=end){ 
		let mid=Math.floor((start + end)/2); 
		if (arr[mid]===number) return true; 
		else if (arr[mid] < number) start = mid + 1; 
		else end = mid - 1; 
	} 

	return false;

} 

File: cpp/binarySearch.cpp

/*

Binary Search algorithm (C++)
By CatProgrammerYT

*/

#include<iostream>
using namespace std;
int binarySearch(int arr[], int p, int r, int num) {
   int mid = (p + r)/2;
   if (p <= r) {
      if (arr[mid] == num)
      return mid ;
      else if (arr[mid] > num)
      return binarySearch(arr, p, mid-1, num);
      else if (arr[mid] > num)
      return binarySearch(arr, mid+1, r, num);
   }
   return -1;
}
int main(void) {
   int arr[] = {1, 3, 7, 15, 18, 20, 25, 33, 36, 40};
   int n = sizeof(arr)/ sizeof(arr[0]);
   int num = 33;
   int index = binarySearch (arr, 0, n-1, num);
   if(index == -1)
   cout<< num <<" is not present in the array";
   else if (index != -1)
   cout<< num <<" is present at index "<< index <<" in the array";
   while (true) {
	   int x;
	   cin >> x;
   } 
   return 0;
}